### PR TITLE
Implemented Velocity Wheel Trajectory Follower

### LINKED ISSUE #1649: Create controller abstraction that takes RPM
### Description of the task
We should have a a controller abstraction that attempts to achieve the given rpm. 

### Acceptance criteria

- [ ] Controller abstraction
- [ ] Testing with robot

### LINKED ISSUE #2044: Implement `velocity_wheels_followPosTrajectory`
### Description of the task
`firmware_robot` has been split into robots that use `force_wheels` and `velocity_wheels`.
To do this, `velocity_wheels_followPosTrajectory` should roughly do the following:
- Convert the position trajectory to a velocity trajectory (Look to the `trajectory_planner` to see how it's done)
- Access the index (which is passed in) to get the relevant velocities
- Do some matrix manipulation to turn it into per wheel velocities and set them to the 4 velocity wheels (using `set_target_velocity`)

### Acceptance criteria

- [ ] Unit tests should be written
- [ ] Manually check with the visualizer
- [ ] Should work with the new simulator protocol changes that will be going in

### PR DESCRIPTION
**Pull request by jonathanlew**
### Description
* Implemented Velocity Wheel Trajectory Follower
* Adjusted the interfaces slightly to add num_elements
* Added setTargetRPM for clarity

### Testing Done
pending

### Resolved Issues
resolves #1649 
resolves #2044

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Review by kelvinkoon — CHANGES_REQUESTED** [review]
LGTM! Just 1 nit.

**Review by jsn0716 — DISMISSED** [review]
LGTM pending Kelvin's comment

**Review by akhilveeraghanta — DISMISSED** [review]
lgtm :100:

File: src/firmware/app/world/velocity_wheel.h [hunk 1]
```diff
@@ -34,16 +34,13 @@ typedef struct VelocityWheelConstants
 /**
  * Create a wheel object with functions for interacting with it
  *
```
[the remark this answers is not in the document]
Comment (jonathanlew) [reply src/firmware/app/world/velocity_wheel.h:36]: fixed

**Review by kelvinkoon — APPROVED** [review]

**Review by jsn0716 — APPROVED** [review]

**Review by akhilveeraghanta — APPROVED** [review]

**Pull request merged**